velox/functions/sparksql/String.h

```diff
@@ -94,6 +94,59 @@ struct Sha1HexStringFunction {
   }
 };
 
+/// sha2 function
+/// sha2(varbinary, bitLength) -> string
+/// Calculate SHA-2 family of functions (SHA-224, SHA-256,
+/// SHA-384, and SHA-512) and convert the result to a hex string.
+/// Returns SHA-2 digest as hex string.
+/// If asking for an unsupported SHA function, the return value is NULL.
+template <typename T>
+struct Sha2HexStringFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE
+  bool call(
+      out_type<Varchar>& result,
+      const arg_type<Varbinary>& input,
+      const int32_t& bitLength) {
+    int digestLength = bitLength >> 3;
+    switch (bitLength) {
+      case 224:
+        result.resize(digestLength * 2);
+        folly::ssl::OpenSSLHash::hash(
+            folly::MutableByteRange((uint8_t*)result.data(), digestLength),
+            EVP_sha224(),
+            folly::ByteRange((const uint8_t*)input.data(), input.size()));
+        break;
+      case 0:
```

Comment:
This branch is confusing. Let's remove it and add logic to check for zero before the switch.